### ISSUE TITLE
VirusTotal results not showing after scan

### ISSUE THREAD

**Issue by zachtu** [objective-see/KnockKnock#31]
The VirusTotal score doesn't seem to be showing after the scan has finished. As shown in the screenshot below, they seem to be stuck on the 'loading' state. This remains even after doing a re-install.

If I export the results to view the score, it looks like this: `"VT detection": "0/0"`.

**System Information**
MacBook Air, M2, 2022
macOS Sonoma 14.2.1

![Screenshot 2024-01-19 at 11 17 47 am](https://github.com/objective-see/KnockKnock/assets/107499062/7e986154-9d92-4a24-94db-8afbce747957)

**Comment by zachtu** [comment]
I have updated the original report with the result of running via the command line.

**Comment by zaqes** [comment]
Same issue here.

**Comment by Samadaeus** [comment]
+1 , I dont have any virus total information showing or an option to submit 

KnockKnock 3.1.0
macOS 15.6

**Comment by JVBarela** [comment]
**VirusTotal queries failing – 401 Unauthorized due to expired partner API key**

Hi 👋,

When running KnockKnock (latest build from GitHub), all VirusTotal queries are failing with **HTTP 401** Unauthorized. The logs show:

`ERROR: failed to query VirusTotal`
`Status Code: 401 (Unauthorized)`
`Domain: NSURLErrorDomain`
`Code: -1012`
`_kCFURLErrorAuthFailedResponseKey = { Status Code = 401, Data Length = 136 }
`

The request is hitting the following endpoint:

`https://www.virustotal.com/partners/sysinternals/file-reports?apikey=233f22e2...
`

From what I can tell, this is a **partner-only VirusTotal API** (Sysinternals). That embedded key now seems to be **revoked or expired**, which explains the 401 errors. The result is that every scanned persistence item comes back as `"VT detection": "0/0"` (no result), even for well-known apps.

**Impact:**

- Local scanning still works fine.
- VirusTotal enrichment is broken.
- End-users may mistakenly read `"0/0"` as “clean,” but it really means “no VT lookup performed.”

**Suggestion:**

- Update KnockKnock to support the **public VirusTotal v3 API** instead of the Sysinternals partner endpoint.
- Allow users to configure their **own VT API key** (via preferences or config file).
- That way, users can still benefit from VT verdicts, even with the free-tier rate limits.

Thanks for maintaining this tool—just wanted to report this so it's on your radar.

**Comment by devahmd** [comment]
This issue stil persist in Novmber 2025

**Comment by objective-see** [comment]
Thanks for your patience. (No project isn't dead, I've just been very busy 🫣).

 TL;DR VT revoked the API key cause other ppl were (ab)using it 😭.

In the latest (pre)released version of KnockKnock, you can now specify your own VT key (which you can get for free from [here](https://docs.virustotal.com/docs/please-give-me-an-api-key). 

Please give it a whirl and let me know if there are any issues:

KK v3.9.5:
https://github.com/objective-see/KnockKnock/releases/tag/v3.9.5
